src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/WeakEventTable.cs

```diff
@@ -297,7 +302,7 @@ private bool Purge(bool purgeAll)
                 {
                     // copy the keys into a separate array, so that later on
                     // we can change the table while iterating over the keys
-                    ICollection ic = _dataTable.Keys;
+                    ICollection<EventKey> ic = _dataTable.Keys;
```

Comment:
Dictionary supports removal during enumeration.  This copying of the keys shouldn't be necessary.

Reply:
No... if you try to remove during enumeration you get: Collection was modified; enumeration operation may not execute.
I think you are confusing ConcurrentDictionary (that can be modified during enumeration, even by other threads) with the plain Dictionary, which cannot be modified during enumeration.

Reply:
You can remove from a dictionary during enumeration ([sharplab](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgAwFgBQiCM65IARyYAsA3OugCICWAxsFQPZQCGYAngDxVTAA0BmJAD48AE1r0mrNngC8eKAFMA7nmp1GLdlx79CwgBQBKdAG90eS3lN5M/AESZ7eAL68LVmwgcJnbj5Y2AMwOQX7uaFbWeEQOROEB0QCsDkkJaC5kaOgAZgxgisw0ABZ4BgBurHgA1uV43GISmtImaOaRVlQ5ZbUAdADSijKy8pitUVHiGlLsvQBKigC2DOWKBn2DbEZZLuRoeQVFpRVV1UP1UI3TWmwDQwDOre1RXT3nI3gI4xOWU5I38yWKzWZy2Oz2hAAnAYAMrAMDcADmvQAUgxuAZ7A5+H9mrNNo9tkA)) but in this case you can't. It removes items from last to first using index, which I don't think is possible without copying the keys to an array. I don't know if there is a valid reason for removing from last to first, other than using index to remove items.

Reply:
You are right... I see this is a difference between the .NET Framework and the .NET Core.